File: src/main/java/dev/thornium/flatcore/registry/FTMaterialIconSet.java
package dev.thornium.flatcore.registry;

import com.gregtechceu.gtceu.api.data.chemical.material.info.MaterialIconSet;

import java.util.HashSet;
import java.util.Set;

public class FTMaterialIconSet {
    public static final Set<MaterialIconSet> ALL = new HashSet<>();

    public static final MaterialIconSet CONSTANTAN = register(new MaterialIconSet("constantan", MaterialIconSet.METALLIC));
    public static final MaterialIconSet PIG_IRON = register(new MaterialIconSet("pig_iron", MaterialIconSet.METALLIC));

    private static MaterialIconSet register(MaterialIconSet iconSet) {
        ALL.add(iconSet);
        return iconSet;
    }

    public static void init() {}
}

File: src/main/kotlin/dev/thornium/flatcore/registry/FTMaterialIconSet.kt
package dev.thornium.flatcore.registry

import com.gregtechceu.gtceu.api.data.chemical.material.info.MaterialIconSet
import dev.thornium.flatcore.api.Initialized

object FTMaterialIconSet : Initialized {
    val ALL = mutableSetOf<MaterialIconSet>()

    val CONSTANTAN = register(MaterialIconSet("constantan", MaterialIconSet.METALLIC))
    val PIG_IRON = register(MaterialIconSet("pig_iron", MaterialIconSet.METALLIC))

    private fun register(iconSet: MaterialIconSet): MaterialIconSet {
        ALL.add(iconSet)
        return iconSet
    }
}
